resources/monitor.rb

```diff
@@ -18,9 +18,19 @@
 
   if Chef::Datadog.agent_major_version(node) != 5
     directory ::File.join(yaml_dir, "#{new_resource.name}.d") do
-      owner 'dd-agent'
-      group 'dd-agent'
-      mode '755'
+      if node['platform_family'] == 'windows'
+        if Chef::Datadog.agent_major_version(node) > 5
```

Comment:
I think this check is not needed, we're already in the `Chef::Datadog.agent_major_version(node) != 5` branch

Reply:
Yes, this is exactly what I was talking about. I hadn't even bother checking the logic, thanks.